iib/workers/tasks/utils.py

```diff
@@ -280,3 +279,311 @@ def _get_function_arg_value(arg_name, func, args, kwargs):
     if arg_value is None and len(args) > arg_index:
         arg_value = args[arg_index]
     return arg_value
+
+
+def get_binary_image_from_config(ocp_version, distribution_scope, binary_image_config={}):
+    """
+    Determine the binary image to be used to build the index image.
+
+    :param str ocp_version: the ocp_version label value of the index image.
+    :param str distribution_scope: the distribution_scope label value of the index image.
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    :return: pull specification of the binary_image to be used for this build.
+    :rtype: str
+    :raises IIBError: when the config value for the ocp_version and distribution_scope is missing.
+    """
+    binary_image = binary_image_config.get(distribution_scope, {}).get(ocp_version, None)
+    if not binary_image:
+        raise IIBError(
+            'IIB does not have a configured binary_image for'
+            f' distribution_scope : {distribution_scope} and ocp_version: {ocp_version}.'
+            ' Please specify a binary_image value in the request.'
+        )
+
+    return binary_image
+
+
+def get_index_image_info(overwrite_from_index_token, from_index=None, default_ocp_version='v4.5'):
+    """
+    Get arches, resolved pull specification and ocp_version for the index image.
+
+    :param str overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for non-privileged users to use
+        ``overwrite_from_index``. The format of the token must be in the format "user:password".
+    :param str from_index: the pull specification of the index image to be resolved.
+    :param str default_ocp_version: default ocp_version to use if index image pull_spec is absent.
+    :return: dictionary of resolved index image pull spec, set of arches, default ocp_version and
+        resolved_distribution_scope
+    :rtype: dict
+    """
+    result = {
+        'resolved_from_index': None,
+        'ocp_version': default_ocp_version,
+        'arches': set(),
+        'resolved_distribution_scope': 'prod',
+    }
+    if not from_index:
+        return result
+
+    with set_registry_token(overwrite_from_index_token, from_index):
+        from_index_resolved = _get_resolved_image(from_index)
+        result['arches'] = _get_image_arches(from_index_resolved)
+        result['ocp_version'] = (
+            get_image_label(from_index_resolved, 'com.redhat.index.delivery.version') or 'v4.5'
+        )
+        result['resolved_distribution_scope'] = (
+            get_image_label(from_index_resolved, 'com.redhat.index.delivery.distribution_scope')
+            or 'prod'
+        )
+        result['resolved_from_index'] = from_index_resolved
+    return result
+
+
+def get_index_image_infos(build_request_config, index_version_map):
```

Comment:
Could you please rename this to `get_index_image_info`

Reply:
Ah, I see there's another method by the same name. Could you please rename this to `get_all_index_image_info` or `get_multiple_index_image_info`

Reply:
Also, could you please add the docstring?